src/object.d

```diff
@@ -3241,6 +3241,40 @@ template RTInfo(T)
     enum RTInfo = null;
 }
 
+bool __equals(T1, T2)(T1[] lhs, T2[] rhs)
```

Comment:
Please add a comment before this function a la `// lhs == rhs lowers to __equals(lhs, rhs) for arrays of all struct types.`